tools/hangar/__snapshots__/test_corpus/valid/for_loop_sequence.w.ts.snap

```diff
@@ -0,0 +1,334 @@
+    {console.log(\`\${i}\`)};
+  }};
+  {console.log(\\"---\\\\nfor i in 0..-2 { ... }\\")};
+  {function* iterator(start, end, inclusive) {
+    let i = start;
+    let limit = inclusive ? ((end < start) ? end - 1 : end + 1) : end;
+    while (i < limit) yield i++;
+    while (i > limit) yield i--;
+  }
+  const iter = iterator(0,(-2),false);
+  for (const i of iter) {
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i <= 0)'\`)})((i <= 0))};
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i > (-2))'\`)})((i > (-2)))};
+    {console.log(\`\${i}\`)};
+  }};
+  {console.log(\\"---\\\\nfor i in 0..=-2 { ... }\\")};
+  {function* iterator(start, end, inclusive) {
+    let i = start;
+    let limit = inclusive ? ((end < start) ? end - 1 : end + 1) : end;
+    while (i < limit) yield i++;
+    while (i > limit) yield i--;
+  }
+  const iter = iterator(0,(-2),true);
+  for (const i of iter) {
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i <= 0)'\`)})((i <= 0))};
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i > (-3))'\`)})((i > (-3)))};
+    {console.log(\`\${i}\`)};
+  }};
+  {console.log(\\"---\\\\nfor i in -2..0 { ... }\\")};
+  {function* iterator(start, end, inclusive) {
+    let i = start;
+    let limit = inclusive ? ((end < start) ? end - 1 : end + 1) : end;
+    while (i < limit) yield i++;
+    while (i > limit) yield i--;
+  }
+  const iter = iterator((-2),0,false);
+  for (const i of iter) {
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i >= (-2))'\`)})((i >= (-2)))};
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i < 0)'\`)})((i < 0))};
+    {console.log(\`\${i}\`)};
+  }};
+  {console.log(\\"---\\\\nfor i in -2..=0 { ... }\\")};
+  {function* iterator(start, end, inclusive) {
+    let i = start;
+    let limit = inclusive ? ((end < start) ? end - 1 : end + 1) : end;
+    while (i < limit) yield i++;
+    while (i > limit) yield i--;
+  }
+  const iter = iterator((-2),0,true);
+  for (const i of iter) {
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i >= (-2))'\`)})((i >= (-2)))};
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i <= 0)'\`)})((i <= 0))};
+    {console.log(\`\${i}\`)};
+  }};
+  const x = 2;
+  {console.log(\\"---\\\\nfor i in 0..x { ... } <=> x = 2\\")};
+  {function* iterator(start, end, inclusive) {
+    let i = start;
+    let limit = inclusive ? ((end < start) ? end - 1 : end + 1) : end;
+    while (i < limit) yield i++;
+    while (i > limit) yield i--;
+  }
+  const iter = iterator(0,x,false);
+  for (const i of iter) {
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i >= 0)'\`)})((i >= 0))};
+    {((cond) => {if (!cond) throw new Error(\`assertion failed: '(i < 2)'\`)})((i < 2))};
+    {console.log(\`\${i}\`)};
+  }};
+  {console.log(\\"---\\\\nfor i in 0..=x { ... } <=> x = 2\\")};
+  {function* iterator(start, end, inclusive) {
```

Comment:
Instead of generating all of this code, would it make more sense to introduce an SDK type (e.g. `std.Range`) which will implement the [the_iterable_protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol). Then, basically the JSification of `a..b` will become `Range.of(a, b)` or something like that.

Reply:
I can use the macro with `let a = std.Range.of(0, 10);` in a Wing file because it generates the macro during compilation, but in the jsification process, I couldn't reference the macro.

but I did a refactoring that I believe improved the generated code, and I kept the `std.Range` class.

Reply:
@marciocadev when you're ready for re-review please click the button next to EladB's name in the Reviewers section of this page (on top right)